IDEAS/Examples/IBPSA/SingleZoneResidentialHydronic.mo

```diff
@@ -56,39 +57,272 @@ model SingleZoneResidentialHydronic
+    y(unit="K"))
     "Block for reading the zone temperature"
     annotation (Placement(transformation(extent={{60,-10},{80,10}})));
   IDEAS.Utilities.IO.SignalExchange.Overwrite TSetExt(
     u(min=273.15+20, max=273.15+80, unit="K"),
     description="Supply temperature set point of the heater")
-    "Block for overwriting control signal" annotation (Placement(transformation(
+    "Block for overwriting supply temperature control signal"
+                                           annotation (Placement(transformation(
         extent={{10,10},{-10,-10}},
         rotation=180,
         origin={10,80})));
   Modelica.Blocks.Sources.Constant TSetConst(k=273.15 + 60)
     "Constant supply temperature set point"
     annotation (Placement(transformation(extent={{-40,70},{-20,90}})));
   Utilities.IO.SignalExchange.Read outputQ(description="Thermal power", KPIs=
-        IDEAS.Utilities.IO.SignalExchange.SignalTypes.SignalsForKPIs.GasPower)
+        IDEAS.Utilities.IO.SignalExchange.SignalTypes.SignalsForKPIs.GasPower,
+    y(unit="W"))
     "Block for outputting the thermal power"
     annotation (Placement(transformation(extent={{60,50},{80,70}})));
   Utilities.IO.SignalExchange.Read outputP(description="Pump electrical power",
-      KPIs=IDEAS.Utilities.IO.SignalExchange.SignalTypes.SignalsForKPIs.ElectricPower)
+      KPIs=IDEAS.Utilities.IO.SignalExchange.SignalTypes.SignalsForKPIs.ElectricPower,
+    y(unit="W"))
     "Block for reading the pump electrical power"
-    annotation (Placement(transformation(extent={{60,20},{80,40}})));
-  Fluid.Actuators.Valves.TwoWayTRV val(
-    redeclare package Medium = Medium,
-    m_flow_nominal=rad.m_flow_nominal,
-    dpValve_nominal=20000,
-    dpFixed_nominal=80000,
-    from_dp=true)
-    annotation (Placement(transformation(extent={{-8,20},{-28,40}})));
+    annotation (Placement(transformation(extent={{60,-40},{80,-20}})));
+
+  Modelica.Blocks.Math.RealToInteger realToInteger
+    annotation (Placement(transformation(extent={{60,-80},{80,-60}})));
+  Controls.Discrete.HysteresisRelease       con(revert=true)
+    "Hysteresis controller for emission system "
+    annotation (Placement(transformation(extent={{-20,-80},{0,-60}})));
+  Modelica.Blocks.Sources.Constant cooSet(k=273.15 + 23, y(unit="K"))
+    "Cooling set point"
+    annotation (Placement(transformation(extent={{-80,-58},{-60,-38}})));
+  Modelica.Blocks.Sources.Constant heaSet(k=273.15 + 21, y(unit="K"))
+    "Heating set point"
+    annotation (Placement(transformation(extent={{-80,-92},{-60,-72}})));
+  Utilities.IO.SignalExchange.Overwrite pumSetExt(u(
+      min=0,
+      max=1,
+      unit="1"), description=
+        "Integer signal to control the stage of the pump either on or off")
+    "Block for overwriting pump control signal" annotation (Placement(
+        transformation(
+        extent={{10,10},{-10,-10}},
+        rotation=180,
+        origin={30,-70})));
+protected
+  model TwoWayOverwriteTRV
```

Comment:
@JavierArroyoBastida this model can be removed?

Reply:
@Mathadon you're right, that's not needed anymore. Removed in https://github.com/open-ideas/IDEAS/pull/1111/commits/90acc02d5156a1c0d0082ce3187d4afa65b8cee1.